packages/@aws-cdk/pipelines/README.md

```diff
@@ -0,0 +1,542 @@
+pipeline.
+
+## Adding Application Stages
+
+To define an application that can be added to the pipeline integrally, define a subclass
+of `Stage`. The `Stage` can contain one or more stack which make up your application. If
+there are dependencies between the stacks, the stacks will automatically be added to the
+pipeline in the right order. Stacks that don't depend on each other will be deployed in
+parallel. You can add a dependency relationship between stacks by calling
+`stack1.addDependency(stack2)`.
+
+Stages take a default `env` argument which the Stacks inside the Stage will fall back to
+if no `env` is defined for them.
+
+An application is added to the pipeline by calling `addApplicationStage()` with instances
+of the Stage. The same class can be instantiated and added to the pipeline multiple times
+to define different stages of your DTAP or multi-region application pipeline:
+
+```ts
+// Testing stage
+pipeline.addApplicationStage(new MyApplication(this, 'Testing', {
+  env: { account: '111111111111', region: 'eu-west-1' }
+}));
+
+// Acceptance stage
+pipeline.addApplicationStage(new MyApplication(this, 'Acceptance', {
+  env: { account: '222222222222', region: 'eu-west-1' }
+}));
+
+// Production stage
+pipeline.addApplicationStage(new MyApplication(this, 'Production', {
+  env: { account: '333333333333', region: 'eu-west-1' }
+}));
+```
+
+### More Control
+
+Every *Application Stage* added by `addApplicationStage()` will lead to the addition of
+an individual *Pipeline Stage*, which is subsequently returned. You can add more
+actions to the stage by calling `addCustomAction()` on it. For example:
+
+```ts
+const testingStage = pipeline.addApplicationStage(new MyApplication(this, 'Testing', {
+  env: { account: '111111111111', region: 'eu-west-1' }
+}));
+
+// Add a custom action -- in this case, a Manual Approval action
+// (for illustration purposes: testingStage.addManualApprovalAction() is a
+// convenience shorthand that does the same)
+testingStage.addCustomAction(new ManualApprovalAction({
+  actionName: 'ManualApproval',
+  runOrder: testingStage.nextSequentialRunOrder(),
+}));
+```
+
+You can also add more than one *Application Stage* to one *Pipeline Stage*. For example:
+
+```ts
+// Create an empty pipeline stage
+const testingStage = pipeline.addStage('Testing');
+
+// Add two application stages to the same pipeline stage
+testingStage.addApplication(new MyApplication1(this, 'MyApp1', {
+  env: { account: '111111111111', region: 'eu-west-1' }
+}));
+testingStage.addApplication(new MyApplication2(this, 'MyApp2', {
+  env: { account: '111111111111', region: 'eu-west-1' }
+}));
+```
+
+## Adding validations to the pipeline
+
+You can add any type of CodePipeline Action to the pipeline in order to validate
+the deployments you are performing.
+
+The CDK Pipelines construct library comes with a `ShellScriptAction` which uses AWS CodeBuild
```

Comment:
Why isn’t this part of the codepipeline actions library?

Reply:
It, or something similar, could be. For now it's not.

Reply:
What are the implications for moving it there? Does it make sense? If yes, let’s open a tracking issue?